frontend/src/components/Editor.vue

```diff
@@ -88,138 +77,105 @@ function toMode(filename: string) {
-  for (const extension of extensions) {
-    const result = extension(filename, editorContent.value, editor, 'hint');
-    if (!result) continue;
-
-    if (result.hint) {
-      CodeMirror.showHint(editor, () => result.hint);
-    }
-  }
-};
-
-const handleLint = (code: string, _: unknown, editor: CodeMirror.Editor) => {
-  const lints: CodeMirror.LintError[] = [];
-
-  for (const extension of extensions) {
-    const result = extension(filename, code, editor, 'lint');
-    if (!result) continue;
+/**
+ * Convert svelte theme name (light/ dark) to Code Mirror theme
+ * @param value svelte theme name
+ */
+function getTheme(value: string): string {
+  return value == 'dark' ? 'dracula' : 'default';
+}
 
-    if (result.lint) {
-      lints.push(...result.lint);
-    }
+let codeMirrorEditor = ref<CodeMirror.EditorFromTextArea | null>(null);
+let editorTag = ref<HTMLTextAreaElement | null>(null);
+
+//initialize codemirror once HTML tag is bound to variable
+watch(editorTag, (newEditorTag) => {
+  if (newEditorTag && !codeMirrorEditor.value) {
+    //https://stackoverflow.com/questions/67686617/codemirror-on-vue3-has-a-problem-when-setvalue-is-kicked
+    codeMirrorEditor.value = markRaw(
+      CodeMirror.fromTextArea(newEditorTag, {
+        mode: toMode(props.filename),
+        autofocus: props.autofocus,
+        lineWrapping: props.wrap,
+        gutters: ['CodeMirror-lint-markers'],
+        spellcheck: true,
+        theme: getTheme(curTheme.value),
+        readOnly: props.disabled,
+        tabSize: 2,
+        extraKeys: {
+          'Ctrl-Space': 'autocomplete',
+          F11: function (cm) {
+            cm.setOption('fullScreen', !cm.getOption('fullScreen'));
+          },
+          Esc: function (cm) {
+            if (cm.getOption('fullScreen')) {
+              cm.setOption('fullScreen', false);
+            }
+          },
+          Tab: function (cm) {
+            cm.execCommand('insertSoftTab');
+          },
+          ...props.extraKeys
+        }
+      })
+    );
+
+    if (modelValue.value) codeMirrorEditor.value.setValue(modelValue.value);
+
+    codeMirrorEditor.value.on('change', (doc) => {
+      modelValue.value = doc.getValue();
+    });
   }
+});
 
-  return lints;
-};
-
-const handleSetup = (editor: CodeMirror.Editor) => {
-  const gutters: string[] = [];
-
-  for (const extension of extensions) {
-    const result = extension(filename, editorContent.value, editor, 'setup');
-
-    if (!result) continue;
-
-    if (result.gutters) {
-      gutters.push(...result.gutters);
-    }
-
-    if (result.spellCheck !== undefined) {
-      editor.setOption('spellcheck', result.spellCheck);
-    }
+function setUpOptions() {
+  if (codeMirrorEditor.value) {
+    codeMirrorEditor.value.setOption('mode', toMode(props.filename));
+    codeMirrorEditor.value.setOption('readOnly', props.disabled);
+    codeMirrorEditor.value.setOption(
+      'gutters',
+      props.filename == '/config.yml' ? ['CodeMirror-lint-markers'] : []
```

Comment:
We spent quite a lot of effort on the old editor to make it generic and not hardcode any filepaths in it. Could you please clarify a bit more about what was broken about it? Can we make smaller fixes to it to make it work?